includes/admin/functions-menu.php

```diff
@@ -50,18 +50,33 @@ function wpas_tickets_count() {
 
 	global $menu, $current_user;
 
-	if ( current_user_can( 'administrator' )
-		 && false === boolval( wpas_get_option( 'admin_see_all' ) )
-		 || ! current_user_can( 'administrator' )
+	$is_agent = current_user_can( 'administrator' )
+		&& false === boolval( wpas_get_option( 'admin_see_all' ) )
+		|| ! current_user_can( 'administrator' )
 			&& current_user_can( 'edit_ticket' )
-			&& false === boolval( wpas_get_option( 'agent_see_all' ) )
-	) {
+			&& false === boolval( wpas_get_option( 'agent_see_all' ) );
 
-		$agent = new WPAS_Member_Agent( $current_user->ID );
-		$count = $agent->open_tickets();
+	$count_cache = get_site_transient( 'wpas_tickets_counts' );
 
-	} else {
-		$count = count( wpas_get_tickets( 'open' ) );
+	if( !is_array($count_cache) )
+		$count_cache = array();
+
+	$agent_id = $is_agent ? $current_user->ID : 0;
+
+	if( !isset( $count_cache[$agent_id] ) )  {
+
+		if ( $is_agent ) {
+
+			$agent = new WPAS_Member_Agent( $agent_id );
+			$count = $agent->open_tickets();
+
+		} else {
+			$count = count( wpas_get_tickets( 'open' ) );
+		}
+
+		$count_cache[$agent_id] = $count;
```

Comment:
Not a fan of using an incorrect user ID as a key of the array. `0`, as set by `$agent_id = $is_agent ? $current_user->ID : 0;` is not a user ID that exists.

Reply:
I did not call it `$user_id`. If the user is not an agent, its `$agend_id` can be 0.
0 is a valid WP user id (its for logged out users).